### ISSUE TITLE
Copying values

### ISSUE THREAD

**Issue by syberant** [splintah/BFAL#1]
Here is some brainfuck code to copy a value from cell A to cell B.

Suppose your cells look like this (italics for pointer):

*A* | B | C
--- | --- | ---
*31* | 0 | 0

The first while loop exhausts A and puts its value in **both** B and C: `[>+>+<<-]`

*A* | B | C
--- | --- | ---
*0* | 31 | 31

Then we move to C: `>>`

A | B | *C*
--- | --- | ---
0 | 31 | *31*

And we put C in A with `[-<<+>>]` to get:

A | B | *C*
--- | --- | ---
31 | 31 | *0*

We should now probably move the pointer back and then we are done.

This functionality would **really** come in handy when you need to use a value multiple times.

**Comment by splintersuidman** [comment]
I will implement a copy function as follows:

### Syntax
```
# Copy the value in cell `a` to cell `b`.
copy a b;
```

### Pseudo-output-code
This is pseudo-code for the outputted Brainfuck, with explicit gotos.
```
create var c;
goto a;
while a {
    goto b;
    incr b;
    goto c;
    incr c;
    goto a;
    decr a;
    goto a; # For the while-loop.
};
goto c;
while c {
    goto a;
    incr a;
    goto c;
    decr c;
    goto c; # For the while loop.
};
delete var c;
```
Because the variable `c` does not exist after executing this, `_nextCell` does not have to be changed (I think).

**Comment by splintersuidman** [comment]
So, I've implemented a copy statement. The code is a mess, but it works well. When a `while`-statement is added, an abstract syntax tree representing the copy algorithm can be constructed, which can be compiled instead of manually adding the Brainfuck algorithm.

**Comment by splintersuidman** [comment]
The `while`-loop AST would look something like this:

```haskell
[ SWhile
  { _name = from
  , _body =
    [ SIncr { _name = to }
    , SIncr { _name = temp }
    , SDecr { _name = from }
    ]
  }
, SWhile
  { _name = temp
  , _body =
    [ SIncr { _name = from }
    , SDecr { _name = temp }
    ]
  }
]
```

There is a problem though: `temp` does not 'have a name'; it is a cell which will be deleted after use.
One way to solve this is by extending `Ident` to be something like this:
```haskell
data Ident
  = Name Ident
  | Cell Int
```
This would allow for, for example, an extension of the language with (unsafe) explicit gotos to random cells, which is not really necessary. So, it seems a bit pointless for other purposes.

**Comment by splintersuidman** [comment]
The idea of the previous comment is implemented in the branch while-ast in commit 8cbcf97.
